Gems/Atom/RHI/Code/Tests/PageTileAllocatorTests.cpp

```diff
@@ -0,0 +1,170 @@
+                if (lastTile < newlastTile)
+                {
+                    lastTile = newlastTile;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        uint32_t GetTileCount(const AZStd::vector<RHI::PageTileSpan>& tilesList)
+        {
+            uint32_t tileCount = 0;
+            for (const auto& tiles : tilesList)
+            {
+                tileCount += tiles.m_tileCount;
+            }
+            return tileCount;
+        }
+    };
+
+    TEST_F(PageTileAllocatorTest, SingleAllocation_Success)
+    {
+        RHI::PageTileAllocator allocator;
+        
+        uint32_t pageTileCount = 256;
+        allocator.Init(pageTileCount);
+
+        ASSERT_TRUE(allocator.GetFreeTileCount() == pageTileCount);
+        ASSERT_TRUE(allocator.GetTotalTileCount() == pageTileCount);
+        ASSERT_TRUE(allocator.GetUsedTileCount() == 0);
+
+        uint32_t allocated = 0;
+        uint32_t requested = 24;
+
+        AZStd::vector<RHI::PageTileSpan> tilesList = allocator.TryAllocate(requested, allocated);
+        ASSERT_TRUE(allocated == requested);
+        ASSERT_TRUE(tilesList.size() == 1);
+        ASSERT_TRUE(tilesList.front().m_tileCount == requested);
+        ASSERT_FALSE(allocator.IsPageFree());
+        
+        ASSERT_TRUE(allocator.GetFreeTileCount() == pageTileCount - requested);
+        ASSERT_TRUE(allocator.GetUsedTileCount() == requested);
+
+        allocator.DeAllocate(tilesList);
+        
+        ASSERT_TRUE(allocator.GetFreeTileCount() == pageTileCount);
+        ASSERT_TRUE(allocator.GetUsedTileCount() == 0);
+        ASSERT_TRUE(allocator.IsPageFree());
+    }
+
+    TEST_F(PageTileAllocatorTest, SingleOutOfRangeAllocation_Failed)
+    {
+        RHI::PageTileAllocator allocator;
+        
+        uint32_t pageTileCount = 20;
+        allocator.Init(pageTileCount);
+
+        uint32_t allocated = 0;
+        uint32_t requested = 24;
+
+        AZStd::vector<RHI::PageTileSpan> tilesList = allocator.TryAllocate(requested, allocated);
+
+        ASSERT_TRUE(allocated == pageTileCount);
+        ASSERT_TRUE(tilesList.front().m_tileCount == allocated);
+        ASSERT_FALSE(allocator.IsPageFree());
+        
+        ASSERT_TRUE(allocator.GetFreeTileCount() == 0);
+        ASSERT_TRUE(allocator.GetUsedTileCount() == pageTileCount);
+
+        allocator.DeAllocate(tilesList);
+        
+        ASSERT_TRUE(allocator.GetFreeTileCount() == pageTileCount);
+        ASSERT_TRUE(allocator.GetUsedTileCount() == 0);
+        ASSERT_TRUE(allocator.IsPageFree());
+    }
+
+    TEST_F(PageTileAllocatorTest, RandomAllocationDeallocation_Success)
+    {
+        RHI::PageTileAllocator allocator;
+        
+        uint32_t pageTileCount = 30;
+        allocator.Init(pageTileCount);
+
+        uint32_t allocationCount = 100;
+
+        AZ::SimpleLcgRandom random(AZStd::GetTimeNowMicroSecond());
```

Comment:
This is fine for running and re-running the test repeatedly locally to get broader coverage, but for AR we should be using a consistent seed so that we have the same results on every test run.